badger/badger.go

```diff
@@ -43,21 +43,36 @@ func NewDatastoreFrom(db *badger.DB) *Datastore {
 }
 
 func (b *Datastore) Get(ctx context.Context, key []byte) ([]byte, error) {
-	txn := b.newTxn(true)
+	txn, ok := ctx.Value(corekv.CtxTxnKey).(*bTxn)
```

Comment:
todo: I think these references to `corekv.CtxTxnKey` need to be pulled out to a shared location - you should be able to include the cast using a limited generic:

```
getTxn[TTxn Txn](ctx context.Context) (TTxn, bool) {
  return ctx.Value(corekv.CtxTxnKey).(TTxn)
}
...
txn, ok := getTxn[*bTxn](ctx)
```

EDIT: I see you already have helpers without the generic.  If you have two functions for each relevant one, you'll let people use the generic-free one when it makes more sense - e.g. `TryGetCtxTxn` and `TryGetCtxTxnG`.